File: easymall/src/main/java/org/ruikun/entity/Category.java
package org.ruikun.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
@TableName("category")
public class Category {
    @TableId(type = IdType.AUTO)
    private Long id;

    private String name;

    private String icon;

    private Long parentId;

    private Integer level;

    private Integer sort;

    private Integer status; // 0-禁用 1-启用

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    @TableLogic
    @TableField(fill = FieldFill.INSERT)
    private Integer deleted;

    @TableField(exist = false)
    private List<Category> children;
}

File: easymall-frontend/src/types/category.ts
export interface CategoryPageItem {
  id: number
  name: string
  icon: string
  parentId: number
  parentName: string
  level: number
  sort: number
  status: number
}

export interface CategoryDetail {
  id: number
  name: string
  icon: string
  parentId: number
  parentName: string
  level: number
  sort: number
  status: number
  createTime: string
  updateTime: string
}

export interface CategoryFormData {
  name: string
  icon: string
  parentId: number | null
  level: number
  sort: number
  status: number
}

export interface CategoryQuery {
  pageNum: number
  pageSize: number
  name?: string
  parentId?: number
  level?: number
  status?: number
}
